runtime/onert/core/src/util/shapeinf/Range.cc

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util/ShapeInference.h"
+#include <cmath>
+namespace onert
+{
+namespace shape_inference
+{
+// helper function
+template <typename T> ir::Shape inferRangeShape(T start_val, T limit_val, T delta_val)
+{
+  ir::Shape out_shape(static_cast<int>(1));
+
+  out_shape.dim(0) =
+      (std::is_integral<T>::value
+           ? ((std::abs(start_val - limit_val) + std::abs(delta_val) - 1) / std::abs(delta_val))
+           : std::ceil(std::abs((start_val - limit_val) / delta_val)));
+  return out_shape;
+}
+
+// StaticInferer at compilation time
+void StaticInferer::visit(const ir::operation::Range &op)
+{
```

Comment:
We have this situation with our target model file:
```
(gdb) p start_op.info().isDynamic()
$9 = false
(gdb) p limit_op.info().isDynamic()
$10 = false
(gdb) p delta_op.info().isDynamic()
$11 = false

(gdb) p start_op.isConstant()
$13 = true
(gdb) p limit_op.isConstant()
$14 = false
(gdb) p delta_op.isConstant()
$12 = true
```

Since they are all static, but one is NOT const, the output will be static (by default). However, I wonder if this is correct. Shouldn't the output be dynamic?

Reply:
I see.  I will apply it. Thanks for your information.

Reply:
Done.